README.md

```diff
@@ -38,6 +38,7 @@ To determine whether to allow response _response_ to a request _request_, run th
 1. If _response_'s status is not an [ok status](https://fetch.spec.whatwg.org/#ok-status), then return false.
 1. If _mimeType_'s essence starts with "`audio/`", "`image/`", or "`video/`", then return false.
 1. If _mimeType_'s essence is "`text/csv`", then return false.
+1. Wait for end-of-file of _response_.
```

Comment:
This probably means that the size of the response is leaked via timing side channels (e.g. see also https://crbug.com/1154250).  OTOH, I am not sure what can be done to fix this problem.

Would it be worth adding a note here to explain why we need to parse the whole response / what would go wrong if we parsed only the first 1024 bytes or so?  AFAIR, the problem is that the first 1024 bytes can look like valid Javascript (e.g. unterminated HTML comment, unterminated JSON list) - unconditionally blocking such responses may block valid Javascript;  unconditionally allowing such responses may disclose cross-site non-script-data.

/cc @csreis

Reply:
Yeah, that seems right. If your JS parser can do things incrementally you could avoid that sometimes perhaps.

It's also that by parsing 1024 bytes your JS parser might not be in a state that makes sense and there's definitely no specification that can handle that scenario currently.